src/demoScreen.c

```diff
@@ -89,8 +95,133 @@ void sub_080A2F8C(void) {
     }
 }
 
-ASM_FUNC("asm/non_matching/demoScreen/sub_080A2FD0.inc", void sub_080A2FD0(void))
+// The condition on the left dpad is very convuluted for some reason (instead oj just an equality check).
+// This causes the entire function heirarchy to change on a whim and I couldn't quite figure out
+// the correct conditions/code to make it like the original function.
+NONMATCH("asm/non_matching/demoScreen/sub_080A2FD0.inc", void sub_080A2FD0(void)) {
+    u32 tmp;
+    u32 tmp2;
+    u32 val;
+    s32 tmp3;
+    u32 unk_0x0;
+    u32 tmp4;
+    u32 keys;
 
-ASM_FUNC("asm/non_matching/demoScreen/sub_080A30AC.inc", void sub_080A30AC(void))
+    if (gFadeControl.active == 0) {
+        val = 0;
+        keys = gInput.heldKeys;
+        if (keys == DPAD_RIGHT) {
+            if (gChooseFileState.unk_0x0 == 0) {
+                val = 1;
+                gChooseFileState.unk_0x20 = 4;
+                SoundReq(SFX_TEXTBOX_CHOICE);
+            }
+        }
+        else if ((keys) > 0x10) {
+            if ((keys) != 0x40) {
+                if ((keys) <= (s16)0x40) {
+                    if (keys == DPAD_LEFT) {
+                        if (gChooseFileState.unk_0x0 == 0) {
+                            val = -1;
+                            gChooseFileState.unk_0x20 = 0xfc;
+                            SoundReq(SFX_TEXTBOX_CHOICE);
+                        }
+                    }
+                }
+            }
+        }
+        else if ((keys) == 0x1 || (keys) == 0x8) {
+            if (gChooseFileState.unk_0x0 == 0) {
+                gMain.screen = 2;
+                gMain.funcIndex = gChooseFileState.unk_0x0;
+                SoundReq(SFX_TEXTBOX_SELECT);
+            }
+        }
+        tmp4 = (0x80 << 0x12);
```

Comment:
this is `gUnk_02000000`

Reply:
But wouldn't it allocate a word in memory if I used it as an extern?

Reply:
It got renamed to gSaveHeader - and yes, it would be - that's why it's actually a macro (defined in `structures.h`).